README.md

```diff
@@ -77,6 +123,21 @@ implement that:
     # (xoauth_principal)
     consumer_principal = access_token.consumer_principal
 
+### Service Provider HMAC-SHA1 Signature Method
+
+The preferred and default signature method is PLAINTEXT, as all communication SHOULD be via TLS. However, if HMAC-SHA1 signatures are necessary, then this can be achieved by passing additional informational to the `authenticate` method.
```

Comment:
Reading through the code I think this is true, but I want to confirm: The agent used on the provider side does **not** need to configure the `signature_method` parameter (i.e. a single agent can authenticate both tokens signed with PLAINTEXT and signed with HMAC-SHA1)?

Reply:
Correct. The authenticate method assumes that the AccessToken instance has the appropriate signature_method and signature values. This should then connect to the from_authorization_header which will pull that out of the token information in the Authorization header.

The only thing the provider needs to do is pass in the extra information to authenticate every time -- the HTTP method, the URI and the request parameters.